.devcontainer/devcontainer.json

```
@@ -1,6 +1,6 @@
 {
     "name": "control libraries",
-    "remoteUser": "developer",
+    "remoteUser": "ubuntu",
```

nitpick: I prefer `containerUser`

```suggestion
    "containerUser": "ubuntu",
```

The `containerUser` is the user to start the container with, while `remoteUser` is the user to exec any processes when attaching to the container. The default `containerUser` is the default user in the Docker image. And, the default `remoteUser` is the `containerUser`. So setting `containerUser` also sets the `remoteUser` implicitly.